### ISSUE TITLE
supress_filters breaks pagination

### ISSUE THREAD

**Issue by TylerBarnes** [wp-graphql/wp-graphql#1259]
If this code is added to a WP instance
```
add_action( 'pre_get_posts', 'mm_show_all_languages_in_graphql' );
function mm_show_all_languages_in_graphql( $query ) {
  if ( function_exists( 'is_graphql_request' ) && true === is_graphql_request() ) {
    $query->query_vars['suppress_filters'] = true;
  }
}
```

Pagination becomes infinite where the start cursor and the end cursor will always be the same, and `hasNextPage` is always true. This makes sense since pagination uses filters, but this leads to broken functionality with no error message.

Notice that the start cursor and end cursor are the same. No matter how many times this is executed it will return the same page.
<img width="1031" alt="Screen Shot 2020-04-22 at 11 50 03 AM" src="https://user-images.githubusercontent.com/14190743/80021480-6d781f00-848f-11ea-8454-94ade80ee1ce.png">

I propose that pagination checks for `suppress_filters`, and it it's set, `hasNextPage` should be false and an error should be thrown.

**Comment by jasonbahl** [comment]
@TylerBarnes thanks for the issue! 

I think using `suppress_filters` should be used very carefully, as filters are very core to how WordPress in general works. 

I don't think folks should be using `suppress_filters` liberally, as it can potentially break many many plugins that filter WP_Query, WPGraphQL included. 

That said, if someone is suppressing filters, WPGraphQL should throw an exception and notify the user _why_ things are not working. 

They can then at least make an educated decision on the tradeoff. Is suppressing filters worth the tradeoff of functionality like GraphQL pagination not working? 🤷‍♂️ 

There's a very good chance it would also break other functionality from other plugins, as it is very common for plugins and themes to filter the WP_Query. 

----

**Action Item**

Detect if `suppress_filters` is applied to WP_Query within WPGraphQL requests and throw an exception explaining the server cannot properly complete the request because of this.

**Comment by jasonbahl** [comment]
closed by #1288
